### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by komakurae**
Login Task:
https://trello.com/c/vj3JOVXV

- added login functionality (getting token and user profile)

### CODE REVIEW

File: lib/models/user/models.dart [hunk 1]
```diff
@@ -1,12 +1,67 @@
...
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'dart:convert';
 
 part 'models.freezed.dart';
 part 'models.g.dart';
 
+UserProfile userProfileFromJson(String str) =>
+    UserProfile.fromJson(json.decode(str));
+
+String userProfileToJson(UserProfile data) => json.encode(data.toJson());
```
Comment (StepanchukPv) [inline lib/models/user/models.dart]: Why do you need those methods?

File: lib/repositories/auth_repository.dart [hunk 2]
```diff
@@ -12,10 +13,20 @@ class AuthRepository {
   Stream<AuthenticationStatus> get authenticationStatus =>
       httpClient.authenticationStatus;
 
+  String get endpointUrl => '/auth';
+
   Future<void> signIn(String userName, String password) async {
-    await Future.delayed(const Duration(seconds: 2));
+    final response = await httpClient.post<DynamicMap>(
+      '$endpointUrl/login',
+      data: <String, String>{
```
Comment (StepanchukPv) [inline lib/repositories/auth_repository.dart:21]: Do not add generic types there, the same for post method

File: lib/repositories/user_repository.dart [hunk 3]
```diff
@@ -2,14 +2,19 @@ import 'package:injectable/injectable.dart';
...
   final HttpClient httpClient;
 
   UserRepository(this.httpClient);
 
-  Future<UserProfile> getUserProfile() async {
-    return UserProfile.fromJson({'userName': 'test'});
+  String get endpointUrl => '/users';
+
+  Future<UserProfile> getUserProfileById(int id) async {
+    final response = await httpClient.get<DynamicMap>('$endpointUrl/$id');
```
Comment (StepanchukPv) [inline lib/repositories/user_repository.dart]: It's no need to add generic types

File: lib/models/user/models.dart (hunk 1 above)
Comment (komakurae) [reply lib/models/user/models.dart]: Oh, I've just missed that from auto-gen https://app.quicktype.io/
my bad

File: lib/repositories/user_repository.dart (hunk 3 above)
Comment (komakurae) [reply lib/repositories/user_repository.dart]: Should I cast it later? Because linter is arguing)

**Review by StepanchukPv — APPROVED** [review]

**Pull request merged**